catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -1323,6 +1350,24 @@ private CswRecordCollection queryProductById(String id) throws CswException {
         return cswRecordCollection;
     }
 
+    private String getRange(String rangeHeader) throws UnsupportedQueryException {
```

Comment:
The biggest advantage to moving all of the checking/parsing logic into this method is that you can confirm that the value passed in is actually a properly formatted numeric representation of a range. Otherwise, I could pass you `bytes=x667xjk1-` and you'd try to parse `x667xjk1`.